syntax/go.vim

```diff
@@ -410,7 +410,7 @@ hi def link     goDeclType          Keyword
 
 " Variable Assignments
 if g:go_highlight_variable_assignments != 0
-  syn match goVarAssign /\v[_.[:alnum:]]+(,\s*[_.[:alnum:]]+)*\ze(\s*\=)/
+  syn match goVarAssign /\v[_.[:alnum:]]+(,\s*[_.[:alnum:]]+)*\ze(\s*(\+|\-|\||\^|\*|\/|\%|\<\<|\>\>|\&|\&\^)?\=[^\=])/
```

Comment:
Using a bracketed list is probably a bit more readable than a subgroup with a lot of `|`s and `\`s?

For example (only tested minimally):

    \v[_.[:alnum:]]+(,\s*[_.[:alnum:]]+)*\ze(\s*[-+%<>!&|^*=]?\=[^\=])

Reply:
That particular example wouldn't quite work, but it's close.  Just for completeness, the `!=` isn't a valid assignment, so there's no ned for `!` in the collection, and that example also doesn't account for `<<=`, `>>=`, and `$^=`.  RE: https://golang.org/ref/spec#Assignments.

However, I was able to reduce some of the branches to a collection.